mne/io/tests/test_pick.py

```diff
@@ -126,3 +128,22 @@ def test_picks_by_channels():
     pick_list2 = _picks_by_type(raw.info, meg_combined=True)
     assert_equal(len(pick_list), len(pick_list2))
     assert_equal(pick_list2[0][0], 'mag')
+
+
+@testing.requires_testing_data
+def test_clean_info_bads():
+
+    data_path = testing.data_path()
+    raw_fname = data_path + '/MEG/sample/sample_audvis_trunc_raw.fif'
+    raw = Raw(raw_fname)
+
+    # select eeg channels excluding bads
+    picks_eeg = pick_types(raw.info, meg=False, eeg=True, exclude='bads')
+    info_eeg = pick_info(raw.info, picks_eeg)
+
+    # select meg channels excluding bads
+    picks_meg = pick_types(raw.info, meg='mag', eeg=False, exclude='bads')
+    info_meg = pick_info(raw.info, picks_meg)
+
+    assert_equal(len(info_eeg['bads']), 0)
+    assert_equal(len(info_meg['bads']), 0)
```

Comment:
Are you sure there are bad channels in the first place?

You should not need testing for this test. Use test_raw.fif in data folder